### PR TITLE
Spir-V: Rescale the frag depth to 0,1 mode when -1,1 mode is used in Vulkan.

### PR DESCRIPTION
**Pull request by FernandoS27**
This fixes Shadows in UE4 games on Vulkan.

### CODE REVIEW

File: src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp [hunk 1]
```diff
@@ -477,6 +477,14 @@ void EmitSetSampleMask(EmitContext& ctx, Id value) {
 }
 
 void EmitSetFragDepth(EmitContext& ctx, Id value) {
+    if (ctx.runtime_info.convert_depth_mode) {
+        const Id type{ctx.F32[1]};
+        const Id unit{ctx.Constant(type, 0.5f)};
```
Comment (ameerj) [inline src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp:482]:
```suggestion
        const Id half{ctx.Const(0.5f)};
```

File: src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp [hunk 2]
```diff
@@ -477,6 +477,14 @@ void EmitSetSampleMask(EmitContext& ctx, Id value) {
...
+        const Id half_depth{ctx.OpFMul(type, value, unit)};
+        const Id new_depth{ctx.OpFAdd(type, half_depth, unit)};
```
Comment (ameerj) [inline src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp:484]: maybe `ctx.OpFma`?

File: src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp [hunk 3]
```diff
@@ -477,6 +477,14 @@ void EmitSetSampleMask(EmitContext& ctx, Id value) {
...
+        ctx.OpStore(ctx.frag_depth, new_depth);
+        return;
+    }
     ctx.OpStore(ctx.frag_depth, value);
```
Comment (ameerj) [inline src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp:488]: Just a nit, I'd invert the condition. This way, only the single instruction is in the body of the condition, and the more complex condition is outside the braces.

Comment (FernandoS27) [reply src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp:488]: sadly it's the most common case.

**Review by ameerj — APPROVED** [review]

**Pull request merged**